### DISCUSSION TITLE
lidr Tree Segmentation issue: Processing time

### DISCUSSION THREAD

**Discussion by ReganKelly** [r-lidar/lidR#698] [Q&A]
Hello,

I am in my Honours Year at RMIT Melbourne and doing my year long project on using Lidar data to identify specific tree species.
I am trying to process my las files through Rstudio using the LidR package to do the following steps:

1. Import Las files
2. Clean noise (lidar data of birds)
3. Decimate the las file to reduce size
4. Normalise
5. Tree segmentation
6. Calculate tree metrics and convex hulls
7. Save and Export

I have been able to successfully do steps 1-4 with the decimated file (using homogenize with a resolution and density of 1 each) being around 303.9 MB in size and the normalised file being 349.5MB.

When I get to the tree segmentation step it works but takes a very long time to process anything. It will take about an hour to get to 20% sometimes (depending on the file).
I was wondering if anyone had any ideas or tips on how to make the tree segmentation processes go any faster?
I have tried differing decimate values and the homogenize(1,1) seems to produce the lowest sized file.

Please let me know if I am doing anything incorrectly or any advice you may have.

Thank you,
Regan

**Comment by Jean-Romain** [comment]
Which algorithm are you using to segment trees ?

**Reply by ReganKelly** [reply]
I am using the segment_trees algorithm to segment my lidr scans.
Here is the code I have been using with nlas1 being the decimated las output:

```r
lasthin1 <- segment_trees(nlas1, li2012())
col <- random.colors(200)
plot(lasthin1, color = "treeID", pal = col)
```

Please let me know if this needs improving

**Reply by Jean-Romain** [reply]
Li's algorithm is inherently slow because it has a worse than quadratic complexity. It means that is you double the number of points you multiply the computation time by more than 4. There is nothing to do except maybe reduce the extent of the point cloud. Processing 4 times  chunks of 500 x 500 m should be faster than one chunk of 1 x 1 km.

**Reply by ReganKelly** [reply]
Ah I see ok that's good to know thank you! Do you know of any other platforms that would be good for tree segmentation? I am currently trying to get a free trial for Lidar360 but am unaware of any other options at this current time.
